library/std/tests/thread.rs

```diff
@@ -14,3 +16,19 @@ fn sleep() {
     thread::sleep(Duration::from_millis(100));
     assert_eq!(*finished.lock().unwrap(), false);
 }
+
+#[test]
+fn sleep_until() {
+    let now = Instant::now();
+    let period = Duration::from_millis(100);
+    let deadline = now + period;
+    thread::sleep_until(deadline);
+
+    let margin = Duration::from_micros(100);
```

Comment:
This margin is way too small, making the test fail in my local test. I think it would be better to just test that the deadline has elapsed, because that is the only guarantee the function makes.

Reply:
good point, ill fix it up tomorrow